src/Compilers/CSharp/Portable/Lowering/AsyncRewriter/AsyncRewriter.cs

```diff
@@ -73,6 +74,31 @@ internal partial class AsyncRewriter : StateMachineRewriter
             return rewriter.Rewrite();
         }
 
+        /// <returns>
+        /// Returns true if all types and members we need are present and good
+        /// </returns>
+        protected bool VerifyPresenceOfRequiredAPIs()
+        {
+            DiagnosticBag bag = DiagnosticBag.GetInstance();
+
+            EnsureWellKnownMember(WellKnownMember.System_Runtime_CompilerServices_IAsyncStateMachine_MoveNext, bag);
+            EnsureWellKnownMember(WellKnownMember.System_Runtime_CompilerServices_IAsyncStateMachine_SetStateMachine, bag);
```

Comment:
Because the C# implementation does extra checks in the `AsyncRewriter` constructor (which I left unchanged).

```C#
            try
            {
                _constructedSuccessfully = AsyncMethodBuilderMemberCollection.TryCreate(F, method, this.stateMachineType.TypeMap, out _asyncMethodBuilderMemberCollection);
            }
            catch (SyntheticBoundNodeFactory.MissingPredefinedMember ex)
            {
                diagnostics.Add(ex.Diagnostic);
                _constructedSuccessfully = false;
            }
```